go/client/fork_server_nix.go

```diff
@@ -8,14 +8,34 @@ package client
 import (
 	"fmt"
 	"os"
+	"os/exec"
 	"syscall"
 
 	"github.com/keybase/client/go/libkb"
 	keybase1 "github.com/keybase/client/go/protocol/keybase1"
+	"github.com/keybase/client/go/systemd"
 )
 
 func spawnServer(g *libkb.GlobalContext, cl libkb.CommandLine, forkType keybase1.ForkType) (pid int, err error) {
 
+	// If we're running under systemd, start the service as a user unit instead
+	// of forking it directly. We do this here instead of higher up, because we
+	// want to handle the case where the service was previously autoforked, and
+	// then the user upgrades their keybase package to a version with systemd
+	// support. The flock-checking code will short-circuit before we get here,
+	// if the service is running, so we don't have to worry about a conflict.
+	if systemd.IsRunningSystemd() {
+		g.Log.Info("Starting keybase.service.")
+		startCmd := exec.Command("systemctl", "--user", "start", "keybase.service")
```

Comment:
does this copy over the environment?

Reply:
(We just talked, in person.) It doesn't, similar to launchd. So the new behavior should be similar to what happens on macOS. Previously we had two different sets of behaviors on Linux:

- If the service was autostarted after boot, it ran with your desktop's environment variables. That wouldn't include anything in e.g. `~/.bashrc`, but it _might_ include `~/.profile`. Unless you're running a Wayland-based desktop, in which case I don't think you even get `~/.profile`.
- If you ran `run_keybase` in your own terminal, then the service would inherit everything.

So this will change to "only inherit session variables like `XDG_CONFIG_HOME` ". I don't feel strongly about whether that's better or worse, but at least it'll be more consistent. (Not to mention consistent with macOS.)